### PR TITLE
feat: add onboarding endpoint and onboarding status to session endpoint

### PR DESCRIPTION
**Pull request by brandonroberts**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR adds processing on onboarding request for the logged in user, and adds the onboarding status to the user session to toggle the onboarding button in the UI.

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  async findOneById (id: number): Promise<DbUser> {
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder.where("id = :id", { id });
+
+    let item: DbUser | null;
+
+    try {
```
Comment (0-vortex) [inline src/user/user.service.ts:28]: Only thing to change:

```
const item = await queryBuilder.getOne();

  if (!item) {
    throw (new NotFoundException);
  }

  return item;
```

File: src/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  async checkAddUser (user: User): Promise<DbUser> {
+    const { user_metadata: { sub: id, user_name } } = user;
+
+    try {
+      const publicUser = await this.findOneById(id as number);
+
+      return publicUser;
+    } catch (e) {
+      // user not found
```
Comment (0-vortex) [inline src/user/user.service.ts:50]: the create function should sit in here, the "sub" contains all the github information we have

**Review by 0-vortex — CHANGES_REQUESTED** [review]
99% LGTM! 👍

**Comment by brandonroberts** [comment]
@0-vortex PR review items resolved

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -17,20 +20,56 @@ export class AuthController {
...
+
+  @Post("/onboarding")
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiOperation({
+    operationId: "updateOnboarding",
+    summary: "Updates onboarding information for user",
+  })
+  @ApiOkResponse({ type: SupabaseAuthDto })
+  @HttpCode(HttpStatus.OK)
```
Comment (0-vortex) [inline src/auth/auth.controller.ts:61]: This is also not really needed based on my current experience

File: src/auth/auth.controller.ts [hunk 4]
```diff
@@ -17,20 +20,56 @@ export class AuthController {
...
       updated_at,
+      is_onboarded: onboarded,
     };
   }
+
+  @Post("/onboarding")
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiOperation({
+    operationId: "updateOnboarding",
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply src/auth/auth.controller.ts:57]: OK, good to know

**Comment by 0-vortex** [comment]
> @0-vortex PR review items resolved

We can merge after this last one! 🚀

**Review by 0-vortex — APPROVED** [review]
LGTM! 👍 

Pushing this through!

**Pull request merged**